confgenerator/confgenerator_test.go

```diff
@@ -125,24 +127,24 @@ func TestGenerateConfsWithValidInput(t *testing.T) {
 	}
 }
 
-func expectedConfig(testName string, validFilePathFormat string, t *testing.T) string {
-	goldenPath := fmt.Sprintf(validFilePathFormat, platform, testName)
-	rawExpectedConfig, err := ioutil.ReadFile(goldenPath)
+func readFileContent(testName string, filePathFormat string, t *testing.T, respectGolden bool) []byte {
+	filePath := fmt.Sprintf(filePathFormat, platform, testName)
+	rawExpectedConfig, err := ioutil.ReadFile(filePath)
 	if err != nil {
-		if *updateGolden {
+		if *updateGolden && respectGolden {
 			// Tolerate the file not found error because we will overwrite it later anyway.
-			return ""
+			return []byte("")
 		} else {
-			t.Fatalf("test %q: error reading the golden conf from %s : %s", testName, goldenPath, err)
+			t.Fatalf("test %q: error reading the file from %s : %s", testName, filePath, err)
 		}
 	}
-	return string(rawExpectedConfig)
+	return rawExpectedConfig
 }
 
-func updateOrCompareGolden(t *testing.T, testName string, expected string, actual string, path string) {
+func updateOrCompareGolden(t *testing.T, testName string, expectedBytes []byte, actual string, path string) {
 	t.Helper()
-	expected = strings.ReplaceAll(expected, "\r\n", "\n")
-	actual = strings.ReplaceAll(actual, "\r\n", "\n")
+	expected := strings.TrimSuffix(strings.TrimSuffix(strings.ReplaceAll(string(expectedBytes), "\r\n", "\n"), "\n"), "\r")
```

Comment:
Not direct relate to this PR but previous changes. The code here seems to deal with the different line ending of Linux and Windows? I think it's no longer needed because now we use lib to detect run-time OS.

Reply:
https://github.com/GoogleCloudPlatform/ops-agent/pull/42 is where it got introduced.
https://github.com/GoogleCloudPlatform/ops-agent/pull/49/files is when we should've removed it

Reply:
Just removed them.

Reply:
Looks like Windows freaked out after the two lines are removed:
* https://github.com/GoogleCloudPlatform/ops-agent/runs/2406160651
* https://github.com/GoogleCloudPlatform/ops-agent/runs/2406167768

On a second thought, whether we use lib to detect run-time OS or not does not affect how line ending is handled, right? Assuming we tend to develop on Linux workstations, it would always trigger the discrepancy of line breakers for `testdata/.../windows` when the tests run on Windows (in this case a Travis Windows container).

Reply:
Test failed because all our windows samples are stilling ending with \n... After PR#49, windows tests is only being trigger if the runtime is windows..

Reply:
we will fix it in followups

Reply:
Got it. I'll leave it as is for this PR then. It does have the benefit that if someone (I meant me) generates the Windows side of golden files on a Linux machine, the line breaker discrepancy do not break tests.